### PR TITLE
Remove timeout

### PR DESCRIPTION
**Pull request by coolo**
To make scheduling our ISOs more reliable, we need to bump timeouts - our API is just not made for short timeouts ;(

### CODE REVIEW

**Comment by aaannz** [comment]
btw. default perl-Net-DBus timeout is 60s. There are places where openQA catch DBus exceptions, but on many it doesn't and I guess if DBus timeout is hit Mojo will report 500 internal server error.

File: lib/OpenQA/Client.pm [hunk 1]
```diff
@@ -52,9 +52,9 @@ sub new {
             last;
         }
     }
-    # When database locking arises, the server takes some time to reply.
-    # We could also adjust sqlite_busy_timeout in DBI server side.
-    $self->inactivity_timeout(40);
+    # Scheduling a couple of hundred jobs takes quite some time - so we better don't
+    # have any timeout at all (default is 20 seconds)
+    $self->inactivity_timeout(0);
```
Comment (bmwiedemann) [inline lib/OpenQA/Client.pm]: maybe just use a large value e.g. 1000, so that nothing gets stuck forever?

Comment (okurz) [inline lib/OpenQA/Client.pm]: Yes, infinite timeout is not good, "very high value" makes sense, e.g. 600 i.e. 10 minutes? It has to finish eventually …

File: etc/apache2/vhosts.d/openqa-common.inc [hunk 2]
```diff
@@ -51,8 +51,8 @@ ProxyPass /error !
 #webSocket proxy pass to different port
 ProxyPass "/api/v1/ws/" "ws://localhost:9527/ws/" keepalive=On
 
-ProxyPass / http://localhost:9526/ keepalive=On
-ProxyPassReverse / http://localhost:9526/
+ProxyPass / http://localhost:9526/ keepalive=On timeout=600
+ProxyPassReverse / http://localhost:9526/ timeout=600
```
Comment (okurz) [inline etc/apache2/vhosts.d/openqa-common.inc]: good but please comment the selection of the value

File: systemd/openqa-webui.service [hunk 3]
```diff
@@ -9,7 +9,7 @@ Requires=openqa-scheduler.service openqa-websockets.service
 # TODO: define whether we want to run the web ui with the same user
 User=geekotest
 Environment="DBUS_STARTER_BUS_TYPE=system"
-ExecStart=/usr/share/openqa/script/openqa prefork -m production --proxy
+ExecStart=/usr/share/openqa/script/openqa prefork -m production --proxy -i 100 -H 400 -w 10
```
Comment (okurz) [inline systemd/openqa-webui.service]: what do these numbers mean? I agree with the change but please add a reasoning

**Comment by aaannz** [comment]
I see that this can be quick fix, but I'm more for properly handling dbus failures and responding to client in timely manner, possible using callbacks for long duration tasks. As is the spirit of Mojo. Instead covering long tasks by higher timeouts.

File: systemd/openqa-webui.service (hunk 3 above)
Comment (aaannz) [inline systemd/openqa-webui.service]: script/openqa prefork --help

```
Options:
  -a, --accepts <number>               Number of connections for workers to
                                       accept, defaults to 1000
  -b, --backlog <size>                 Listen backlog size, defaults to
                                       SOMAXCONN
  -c, --clients <number>               Maximum number of concurrent
                                       connections, defaults to 1000
  -G, --graceful-timeout <seconds>     Graceful timeout, defaults to 20.
  -I, --heartbeat-interval <seconds>   Heartbeat interval, defaults to 5
  -H, --heartbeat-timeout <seconds>    Heartbeat timeout, defaults to 20
  -h, --help                           Show this summary of available options
      --home <path>                    Path to home directory of your
                                       application, defaults to the value of
                                       MOJO_HOME or auto-detection
  -i, --inactivity-timeout <seconds>   Inactivity timeout, defaults to the
                                       value of MOJO_INACTIVITY_TIMEOUT or 15
  -l, --listen <location>              One or more locations you want to
                                       listen on, defaults to the value of
                                       MOJO_LISTEN or "http://*:3000"
  -M, --multi-accept <number>          Number of connections to accept at
                                       once, defaults to 50
  -m, --mode <name>                    Operating mode for your application,
                                       defaults to the value of
                                       MOJO_MODE/PLACK_ENV or "development"
  -P, --pid-file <path>                Path to process id file, defaults to
                                       "prefork.pid" in a temporary diretory
  -p, --proxy                          Activate reverse proxy support,
                                       defaults to the value of
                                       MOJO_REVERSE_PROXY
  -r, --requests <number>              Maximum number of requests per
                                       keep-alive connection, defaults to 25
  -w, --workers <number>               Number of workers, defaults to 4
```

File: lib/OpenQA/Client.pm (hunk 1 above)
Comment (aaannz) [inline lib/OpenQA/Client.pm]: Well, once apache (after 600s) or sooner openqa (after 400s) closes the connection then client stops waiting.

File: etc/apache2/vhosts.d/openqa-common.inc (hunk 2 above)
Comment (coolo) [inline etc/apache2/vhosts.d/openqa-common.inc]: "I picked 600" is a great comment

**Comment by coolo** [comment]
dbus is not failing, it just takes very long ;(

**Comment by coolo** [comment]
merging this now to get our production system under control again

**Pull request merged**